### PR TITLE
Login improvements

### PR DESCRIPTION
**Pull request by DeanLogan**
Description: This pull request represents the changes made to login security. This includes creating a new page for the login, along with adding additional protection against XSS attacks by adding CSRF tokens, adding a Captcha for the login along with 2FA through an OTP. A "forgot password" button has also been added that will allow users to send an email to the admin notifying them that a new password is needed for their account.

Changes:

- [Added the Captcha so that users had to prove they are not a bot to login](https://github.com/KyleMcComb/Pathfinder/commit/c8cb56ad4740b7102acae48b500e1089eaff0808)
- [Added OTP libraries](https://github.com/KyleMcComb/Pathfinder/commit/ca241c8c360cd2ca310625d7c979ca14d00e01a3)
- [Added QR code to add account to users authentication app of choice](https://github.com/KyleMcComb/Pathfinder/commit/daf78f79113ee1b574e4e7ef8a30a3a50c7c2773)
- [Finished OTP completely, added a new TOTP Device table to the database](https://github.com/KyleMcComb/Pathfinder/commit/b4d0166c07693e7623c1edd8fc590ec7d8489e68)
- [Removed login pop up and replaced with new login page UI](https://github.com/KyleMcComb/Pathfinder/commit/61f19d91ecb62cf3de7a4cf0592ddff87ed7585b)
- [Added forgot email link, which sends an email to the admin notifying them of the user that has forgotten their password](https://github.com/KyleMcComb/Pathfinder/commit/a5a7073ebed7ac8939b39e6d5f7eb94f0003523e)

### CODE REVIEW

File: database/migrations/0001_initial.py
Comment (Rossmcal) [inline database/migrations/0001_initial.py:1]: Line 96. This table isn't going to exist anymore, I am not sure if it should be removed now or after the adding data branch has been merged.

Comment (Rossmcal) [reply database/migrations/0001_initial.py:1]: Actually ignore this as it is in the migrations folder.

**Review by KyleMcComb — APPROVED** [review]
Looks good to me. Good use of comments. Everything makes sense.

**Review by conornugent1 — APPROVED** [review]
Easy to follow and well written code

**Pull request merged**